frontend/src/components/NavBar.js

```diff
@@ -23,15 +25,30 @@ const NavBar = () => {
   )
 }
 
+const Logo = forwardRef(function LogoWithRef({ onClick, href }, ref) {
+  return (
+    <a style={{ display: 'flex' }} href={href} onClick={onClick} ref={ref}>
+      <Image
+        width={160}
+        height={46}
+        objectFit="scale-down"
+        className={styles.logoImg}
+        src={logoSrc}
+        alt="logo"
+      />
+    </a>
+  )
+})
+
```

Comment:
Huh? forwardRef? What's going on here? Why not just use `Link` instead of `a`?

Reply:
This is wrapped with `Link`
https://github.com/kitspace/kitspace-v2/blob/df88c9bbde9cb32efc4c9ec71424679ba66fa38b/frontend/src/components/NavBar.js#L72-L74
`Link` passes a ref to its child, if the child is a functional commponent it has to be wrapped with forwardRef.

Reply:
But why? It seems kind of complicated for what it's doing. There must be a simpler way...

Reply:
I couldn't find any other solution. Maybe use `a` but we lose the features of the `Link`.

Reply:
Why use an `a` at all? Why not put the `Link` where the `a` is?

Reply:
This is the original code 
```jsx
<Link href="/" passHref>
  <Image
    width={160}
    height={46}
    objectFit="scale-down"
    className={styles.logoImg}
    src={logoSrc}
    alt="logo"
  />
</Link>
```
It causes this error in the browser, it's a runtime error, not eslint one
```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
```
If a `Link` is used in the `Logo` component
https://github.com/kitspace/kitspace-v2/blob/df88c9bbde9cb32efc4c9ec71424679ba66fa38b/frontend/src/components/NavBar.js#L28-L42
we will end up in the same situation just introduced some redundant wrapper.

Reply:
Why do we need `passHref`? I don't really understand why this is getting so complicated, all we want is an image with a link to `/`.

Reply:
Because using `Link` without `passHref` will work but not totally, for example, if you hover over the image it won't show the link of `/` at the bottom which works with the plain `a`. As I said before the simplest solution would be using
```jsx
<a href="/" >
  <Image
    width={160}
    height={46}
    objectFit="scale-down"
    className={styles.logoImg}
    src={logoSrc}
    alt="logo"
  />
</a>
```
and disable eslint rule for this line.

Reply:
Does `Link` never show the url at the bottom? If we use an `a` it will do a full refresh/navigation when you click on it which seems like a waste. If you can't find a way to clean it up please leave a comment with a reminder to clean it up down the line and an explanation of why it's like this.

Reply:
> Does Link never show the url at the bottom?

It works if the `passHref` prop is `true`.